restapi/src/main/java/io/stargate/web/docsapi/dao/Paginator.java

```diff
@@ -0,0 +1,117 @@
+package io.stargate.web.docsapi.dao;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.stargate.db.datastore.Row;
+import io.stargate.web.docsapi.exception.DocumentAPIRequestException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+
+/**
+ * This class is in charge of keeping the data of the Docs API pagination process.
+ * It's not only a stateful component, but a state machine with a initial state.
+ * It's created on the REST layer and passed through to the DocumentService that
+ * is in charge of its changes. After being populated its `documentPageState`
+ * can be passed by the REST layer to the client as a JSON string.
+ */
+public class Paginator {
+
+  private final ObjectMapper mapper = new ObjectMapper();
+
+  public final int dbPageSize;
+  public final int docPageSize;
+
+  private boolean initialState = true;
+  private ByteBuffer currentDbPageState; // keeps track of the DB page state while querying the DB
+  private DocumentSearchPageState documentPageState = null;
+
+  public Paginator(String pageStateParam, int pageSizeParam, int dbPageSize) throws IOException {
+    docPageSize = Math.max(1, pageSizeParam);
+    if (docPageSize > DocumentDB.MAX_PAGE_SIZE) {
+      throw new DocumentAPIRequestException("The parameter `page-size` is limited to 20.");
+    }
+
+    this.dbPageSize = dbPageSize;
```

Comment:
Is this parameter used? I could not find places where it is requested.

Reply:
This is odd. I have used in some places. Maybe It was left behind and not committed